cosmos/dbt/graph.py

```diff
@@ -399,8 +399,24 @@ def load_from_dbt_manifest(self) -> None:
         with open(self.project.manifest_path) as fp:  # type: ignore[arg-type]
             manifest = json.load(fp)
 
+            # Validate expected structure in manifest
+            if not all(key in manifest for key in ["nodes", "sources", "exposures"]):
+                logger.error("Manifest file structure is not as expected.")
+                raise CosmosLoadDbtException("Invalid structure in manifest file")
```

Comment:
wondering what happens today if this validation isn't in place when you have an incorrect manifest?